Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -147,106 +115,15 @@ void PulseHeightTrackMatch::startOfCycle()
-        continue;
-      }
-
-      for (int iTrack = triggerTracks.getFirstTrack(); iTrack < triggerTracks.getFirstTrack() + triggerTracks.getNumberOfTracks(); iTrack++) {
-
-        const auto& track = tracks[iTrack];
-        int ntracklets = track.getNtracklets();
-        mTrackletsPerMatchedTrack->Fill(ntracklets);
-
-        for (int iLayer = 0; iLayer < 6; iLayer++) {
-          int trackletIndex = track.getTrackletIndex(iLayer);
-
-          if (trackletIndex == -1) {
-            mDebugCounter->Fill(5);
-            continue;
-          }
-
-          const auto& trklt = tracklets[trackletIndex];
-
-          // obtain pad number relative to MCM center
-          // int padLocal = trklt.getPositionBinSigned() * o2::trd::constants::GRANULARITYTRKLPOS;
-          // MCM number in column direction (0..7)
-          // int mcmCol = (trklt.getMCM() % o2::trd::constants::NMCMROBINCOL) + o2::trd::constants::NMCMROBINCOL * (trklt.getROB() % 2);
-          // int col = 6.f + mcmCol * ((float)o2::trd::constants::NCOLMCM) + padLocal;
-
-          int coldif;
-          for (int iDigit = trigger.getFirstDigit() + 1; iDigit < trigger.getFirstDigit() + trigger.getNumberOfDigits() - 1; ++iDigit) //+1 and -1 for finding the consecutive digits
-          {
-            const auto& digit = digits[iDigit];
-            if (digit.getDetector() != trklt.getDetector()) {
-              mDebugCounter->Fill(6);
-              continue;
-            }
-            if (digit.getPadRow() != trklt.getPadRow()) {
-              mDebugCounter->Fill(7);
-              continue;
-            }
-            coldif = digit.getPadCol() - trklt.getPadCol();
-            if (TMath::Abs(coldif) > 1) {
-              mDebugCounter->Fill(8);
-              continue; // PadCol may be off by +-1 due to the slope of tracklet
-            }
-            const auto& digitBefore = digits[iDigit - 1];
-            const auto& digitAfter = digits[iDigit + 1];
-
-            bool consecutive = false;
-            if (digitBefore.getDetector() == digitAfter.getDetector() && digitBefore.getPadRow() == digit.getPadRow() && digit.getPadRow() == digitAfter.getPadRow() && digit.getPadCol() + 1 == digitBefore.getPadCol() && digitAfter.getPadCol() + 1 == digit.getPadCol()) { // looking for consecutive digits in space
-              mDebugCounter->Fill(9);
-              consecutive = true;
-            }
-
-            if (consecutive) {
-              mDebugCounter->Fill(10);
-              int phVal = 0;
-              for (int iTb = 0; iTb < o2::trd::constants::TIMEBINS; ++iTb) {
-                phVal = digitBefore.getADC()[iTb] + digit.getADC()[iTb] + digitAfter.getADC()[iTb];
-                mPulseHeightpro->Fill(iTb, phVal);
-                mPulseHeightperchamber->Fill(iTb, digit.getDetector(), phVal);
-              }
-            }
-          }
-        }
-      }
+  for (const auto& phData : phDataArr) {
+    // fill for ITS,TPC,TRD tracks
+    if (phData.getType() == 0) {
```

Comment:
could you make the type a configurable parameter? By default I would use everything as input that we have, but we should be able to select a certain type only

Reply:
thanks. yes. i checked, it seems that value of the parameter is only 0 or 1. should keep 1 as default??

Reply:
the parameter can be anything from 0 to 3. By default everything should be accepted. But one should be able to select any combination of the parameters. So basically a bitmap would be good which has all 4 bits set by default